dask_ml/metrics/regression.py

```diff
@@ -16,7 +17,9 @@ def _check_sample_weight(sample_weight: Optional[ArrayLike]):
 def _check_reg_targets(
     y_true: ArrayLike, y_pred: ArrayLike, multioutput: Optional[str]
 ):
-    if multioutput is not None and multioutput != "uniform_average":
+    if multioutput is not None and (
+        is_dask_collection(multioutput) or multioutput != "uniform_average"
+    ):
```

Comment:
Not immediately clear to me why we need this extra check? Can there be a case where `multioutput` is a Dask collection that is also equal to `"uniform_average"`?

Reply:
This is an argument for [sklearn.metric.r2_score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.r2_score.html#sklearn-metrics-r2-score), where `multioutput` can be a str or array-like. In Dask-ML there is [a test](https://github.com/dask/dask-ml/blob/main/tests/metrics/test_regression.py#L113) that specifically passes a Dask Array to `multioutput` and it was failing because Dask collections are ambiguous for truthy.